python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py

```diff
@@ -305,6 +228,30 @@ def _overlay_her_flood_lists(self: AvdStructuredConfigNetworkServicesProtocol) -
 
         return overlay_her_flood_lists
 
-    @cached_property
-    def _multi_vtep(self: AvdStructuredConfigNetworkServicesProtocol) -> bool:
-        return self.shared_utils.mlag is True and self.shared_utils.evpn_multicast is True
+    def _check_for_duplicates(
+        self: AvdStructuredConfigNetworkServicesProtocol,
+    ) -> None:
+        """
+        Raise an exception if the passed vlan or VRF object VNI is already in the vnis dictionary.
+
+        If no duplicate is found, add the vlan or vrf object to the vnis dictionary
+        """
+        vnis: dict[int, EosCliConfigGen.VxlanInterface.Vxlan1.Vxlan.VrfsItem | EosCliConfigGen.VxlanInterface.Vxlan1.Vxlan.VlansItem] = {}
+        for vlan in self.structured_config.vxlan_interface.vxlan1.vxlan.vlans:
+            if vlan.vni in vnis:
```

Comment:
I think it will be more efficient to use defaultdict and append the object and then walk the list in the end looking for more than one. That way you avoid iterating over the growing list every time